src/LobbyServer.hs

```diff
@@ -406,3 +420,35 @@ getClientName remoteClientList = do
   clientList <- liftIO $ CC.readMVar concurrentClientList
   let client = find ((sid ==) . sessionID) clientList
   return $ name $ fromJust client
+
+-- |Sets the password (as a 'ByteString') of the game the client is in.
+-- |Only possible if the client is the owner of the game.
+setPasswordToGame :: Server GamesList -> String -> Server ()
+setPasswordToGame remoteGames passwordString = do
+  let password = pack passwordString
+  hashedPassword <- liftIO $ makePassword password 17
+  mVarGames <- remoteGames
+  maybeGame <- findGameWithSid mVarGames
+  ownerOfGame <- isOwnerOfGame remoteGames
+  case (maybeGame, ownerOfGame) of
+    (Just game, True)          -> liftIO $
+      CC.modifyMVar_ mVarGames $ \games ->
+        return $ updateListElem
+          (\(guuid, gData) -> (guuid, gData {gamePassword = hashedPassword}))
+          (== game)
+          games
+    (Just (_,gameData), False) -> do
```

Comment:
What is happening here. Get sid. Don't care if ClientEntry is found, `return ()` either way. Just `return ()` directly.